astroquery/xmm_newton/core.py

```diff
@@ -133,8 +134,16 @@ def download_data(self, observation_id, filename=None, verbose=False,
         link = link + "".join("&{0}={1}".format(key, val)
                               for key, val in kwargs.items())
 
-        response = self._request('GET', link, save=True)
-        if response is not None:
+        if verbose:
+            log.info(link)
+
+        response = None
+        try:
+            response = self._request('GET', link, save=True, cache=True)
+        except HTTPError:
+            log.error("Unable to access URL {0}".format(link))
```

Comment:
We prefer to use `requests`' built-in error handling:
```suggestion
        response = self._request('GET', link, save=True, cache=True)
        response.raise_for_error()
```

Reply:
From what I can see, `response = self._request('GET', link, save=True, cache=True)` returns a string and not an HTTP response.
Am I right?

Reply:
yes, with `save=True`, the returned item is the path to the downloaded file